sdks/python/apache_beam/runners/dataflow/internal/dependency.py

```diff
@@ -99,173 +91,47 @@
-            '--extra_packages command line option.' % package)
+        # Source is a GCS file but target is local file.
+        with gcsio.GcsIO().open(from_path, mode='rb') as g:
+          with open(to_path, 'wb') as f:
+            pfun = functools.partial(g.read, gcsio.DEFAULT_READ_BUFFER_SIZE)
+            for chunk in iter(pfun, ''):
+              f.write(chunk)
     else:
-      local_packages.append(package)
-
-  if staging_temp_dir:
-    local_packages.extend(
-        [FileSystems.join(staging_temp_dir, f) for f in os.listdir(
-            staging_temp_dir)])
-
-  for package in local_packages:
-    basename = os.path.basename(package)
-    staged_path = FileSystems.join(staging_location, basename)
-    file_copy(package, staged_path)
-    resources.append(basename)
-  # Create a file containing the list of extra packages and stage it.
-  # The file is important so that in the worker the packages are installed
-  # exactly in the order specified. This approach will avoid extra PyPI
-  # requests. For example if package A depends on package B and package A
-  # is installed first then the installer will try to satisfy the
-  # dependency on B by downloading the package from PyPI. If package B is
-  # installed first this is avoided.
-  with open(os.path.join(temp_dir, EXTRA_PACKAGES_FILE), 'wt') as f:
-    for package in local_packages:
-      f.write('%s\n' % os.path.basename(package))
-  staged_path = FileSystems.join(staging_location, EXTRA_PACKAGES_FILE)
-  # Note that the caller of this function is responsible for deleting the
-  # temporary folder where all temp files are created, including this one.
-  file_copy(os.path.join(temp_dir, EXTRA_PACKAGES_FILE), staged_path)
-  resources.append(EXTRA_PACKAGES_FILE)
-
-  return resources
-
-
-def _get_python_executable():
-  # Allow overriding the python executable to use for downloading and
-  # installing dependencies, otherwise use the python executable for
-  # the current process.
-  python_bin = os.environ.get('BEAM_PYTHON') or sys.executable
-  if not python_bin:
-    raise ValueError('Could not find Python executable.')
-  return python_bin
-
-
-def _populate_requirements_cache(requirements_file, cache_dir):
-  # The 'pip download' command will not download again if it finds the
-  # tarball with the proper version already present.
-  # It will get the packages downloaded in the order they are presented in
-  # the requirements file and will not download package dependencies.
-  cmd_args = [
-      _get_python_executable(), '-m', 'pip', 'download', '--dest', cache_dir,
-      '-r', requirements_file,
-      # Download from PyPI source distributions.
-      '--no-binary', ':all:']
-  logging.info('Executing command: %s', cmd_args)
-  processes.check_call(cmd_args)
+      # Branch used only for unit tests and integration tests.
+      # In such environments GCS support is not available.
+      if not os.path.isdir(os.path.dirname(to_path)):
+        logging.info(
```

Comment:
<!--new_thread; commit:d2647e7c0c6001a89be4dc0a673e17d5ce0d951a; resolved:0-->
Can we defer to superclass implementaion for this branch?